submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,300 @@
+will be used. Also, we will internally use configMaps to maintain the state of allocated GlobalIPs
+and get rid of annotations on the objects. Because of this, backward compatibility cannot be maintained
+with the existing implementation of Globalnet.
+
+### User Stories
+
+1.User has two (or more) clusters with Overlapping CIDRs and requires the Pods in one
+  Cluster be able to discover and talk to exported Services.
+
+  To achieve this use-case, user can simply deploy Submariner Globalnet on all the participating
+  clusters. By default, a globalIP (or a set of globalIPs) will be assigned at the Cluster level
+  which will be used as egressIPs for inter-cluster communication and necessary Services can be
+  exported.
+
+2.In a Globalnet deployment user wants all the Pods in a namespace `ns1` to use a unique globalIP
+  instead of the globalIP assigned at the cluster level.
+
+  To achieve this, one can apply the following CRD in the namespace `ns1`.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalEgressIP
+   metadata:
+     name: ns-egressip
+     namespace: ns1
+   spec:
+     numGlobalIPs: 1
+```
+
+3.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIPs as egressIP for cross-cluster communication.
+
+  To achieve this, one can apply the following CRD in the namespace `ns1`.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalEgressIP
+   metadata:
+     name: db-pods
+     namespace: ns1
+   spec:
+     podSelector:
+         matchLabels:
+           role: db
+     NumGlobalIPs: 2
+```
+
+5.In a Globalnet deployment, user wants a specific service to be made available to remote clusters.
+  For this, user can create a ClusterIP Service and export the Service. Globalnet will allocate a
+  globalIP for ingress connectivity such that any external applications can connect to this Service.
+
+```shell
+  kubectl -n default create deployment nginx --image=nginxinc/nginx-unprivileged:stable-alpine
+  kubectl -n default expose deployment nginx --port=8080
+  subctl export service --namespace default nginx
+```
+
+6.In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
+  to use a unique globalIP both as ingress and egressIP for cross-cluster communication.
+
+  To achieve this, user can create a Headless Service (backed by a Deployment or StatefulSets) and
+  Globalnet will ensure that all the backend Pods that belong to the Headless Service are assigned
+  a globalIP. User is also expected **not** to create any GlobalEgressIP object which selects
+  the backendPods that match the Headless Service.
+
+![HeadlessService](./images/globalnet-headless-svc.png)
+
+  In the above example, each of the http pods will get its own globalIP which can be used for
+  both ingress and egress communication.
+
+  However, if the user wants egress traffic from the http Pods to carry a unique globalIP,
```

Comment:
This conflicts with the order stated in the diagram.

Reply:
No, it doesn't. Are you referring to `/images/globalnet-svc.png` at L271 image or something else?